reliability/single-fit/okhttp/src/test/kotlin/com/datadog/android/okhttp/HeadBasedSamplingTest.kt

```diff
@@ -341,24 +345,20 @@ class HeadBasedSamplingTest {
         val eventsWritten = stubSdkCore.eventsWritten(Feature.TRACING_FEATURE_NAME)
         assertThat(eventsWritten).hasSize(2)
 
-        val localSpanPayload = JsonParser.parseString(eventsWritten[0].eventData).asJsonObject
+        val localSpanPayload = JsonParser.parseString(eventsWritten.first().eventData).asJsonObject
         SpansPayloadAssert.assertThat(localSpanPayload)
             .hasSpanAtIndexWith(0) {
                 hasLeastSignificant64BitsTraceId(
                     leastSignificantTraceId.toLong().toHexString().padStart(16, '0')
                 )
                 hasMostSignificant64BitsTraceId(mostSignificantTraceId)
-                hasParentId("0")
-                hasAgentPsr(1.0)
-                hasSamplingPriority(PrioritySampling.SAMPLER_KEEP)
+                hasParentId("0000000000000000")
+                hasRulePsr(1.0)
+                hasSamplingPriority(DatadogTracingConstants.PrioritySampling.USER_KEEP)
```

Comment:
why we now are having `USER_KEEP` instead of `SAMPLER_KEEP`?

Reply:
Because CoreTracer operates slightly different from AndroidTracer.